### PR TITLE
fix(sec): validate page and limit api parameters

### PR DESCRIPTION
**Pull request by kduret**
## Description

validate page and limit api parameters

**Fixes** MON-10854

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 20.04.x
- [x] 20.10.x
- [x] 21.04.x
- [x] 21.10.x (master)

### CODE REVIEW

**Review by adr-mo — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by adr-mo — DISMISSED** [review]

File: src/Centreon/Domain/RequestParameters/RequestParametersException.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * This class is designed to contain all exceptions for the context of the request parameters.
+ *
+ * @package Centreon\Domain\RequestParameters
+ */
+class RequestParametersException extends \InvalidArgumentException
+{
+    /**
+     * @param string $paremeterName
```
Comment (sc979) [inline src/Centreon/Domain/RequestParameters/RequestParametersException.php:33]:
```suggestion
     * @param string $parameterName
```

**Review by adr-mo — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -16207,3 +16207,6 @@ msgstr "L'année doit être inférieure à %d (%s)"
 
 msgid "[%s] The date \"%s\" was expected to be at most %s"
 msgstr "[%s] La date \"%s\" devait être au maximum au %s"
+
+msgid "The parameter \"%s\" must be an integer"
+msgstr "Le paramètre \"%s\" doit ête un entier"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16212]:
```suggestion
msgstr "Le paramètre \"%s\" doit être un entier"
```

File: src/EventSubscriber/CentreonEventSubscriber.php [hunk 3]
```diff
@@ -187,10 +188,22 @@ public function initRequestParameters(RequestEvent $request): void
     {
         $query = $request->getRequest()->query->all();
 
-        $limit = (int) ($query[RequestParameters::NAME_FOR_LIMIT] ?? RequestParameters::DEFAULT_LIMIT);
+        $limit = filter_var(
+            $query[RequestParameters::NAME_FOR_LIMIT] ?? RequestParameters::DEFAULT_LIMIT,
+            FILTER_VALIDATE_INT
+        );
+        if (empty($limit)) {
```
Comment (sc979) [inline src/EventSubscriber/CentreonEventSubscriber.php:195]:
```suggestion
        if (false === $limit) {
```

File: src/EventSubscriber/CentreonEventSubscriber.php [hunk 4]
```diff
@@ -187,10 +188,22 @@ public function initRequestParameters(RequestEvent $request): void
...
+            throw RequestParametersException::integer(RequestParameters::NAME_FOR_LIMIT);
+        }
         $this->requestParameters->setLimit($limit);
 
-        $page = (int) ($query[RequestParameters::NAME_FOR_PAGE] ?? RequestParameters::DEFAULT_PAGE);
+        $page = filter_var(
+            $query[RequestParameters::NAME_FOR_PAGE] ?? RequestParameters::DEFAULT_PAGE,
+            FILTER_VALIDATE_INT
+        );
+        if (empty($page)) {
```
Comment (sc979) [inline src/EventSubscriber/CentreonEventSubscriber.php:204]:
```suggestion
        if (false === $page) {
```

**Review by sc979 — DISMISSED** [review]
As discussed, I think that "false" should be checked. But as if the variable is "false" it also will returns empty == true, I'm approving

**Review by sc979 — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**